### DISCUSSION TITLE
Rewrite in middleware discards empty query params

### DISCUSSION THREAD

**Discussion by jacobhallenborgprototyp** [vercel/next.js#51023] [Help]
### Summary

When rewriting a request in the `middleware.ts`, then the query params with no value are discarded.

```js
NextResponse.rewrite('https://example.com?param1=hello&param2=')
```

The server [example.com]() receives only `?param1=hello` and the `param2` is dropped for some reason.

This happens in my Vercel deployments, but not locally when running with `next dev` or `next start`

### Additional information

```js
NextJS version: 13.4.4
Hosting: Vercel
```

**Comment by Roberto14** [comment]
I'm having a similar issue on next 12.3.4. Trying to match a rewrite on an empty query param as follows doesn't seem to work:
```
{
  source: '/pageX/:path*',
    has: [
      {
        type: 'query',
        key: 'param1'
      }
    ],
  destination: '/pageY/:path*'
}
```
The call to `/pageX/a/b/c?param1` ideally should rewrite to `/pageY/a/b/c` but it won't.
